modules/xmpp/JingleSessionPC.js

```diff
@@ -1799,6 +1894,113 @@ export default class JingleSessionPC extends JingleSession {
     }
 
     /**
+     * Will put and execute on the queue a session modify task. Currently it
+     * only checks the senders attribute of the video content in order to figure
+     * out if the remote peer has video in the inactive state (stored locally
+     * in {@link _remoteVideoActive} - see field description for more info).
+     * @param {jQuery} jingleContents jQuery selector pointing to the jingle
+     * element of the session modify IQ.
+     * @see {@link _remoteVideoActive}
+     * @see {@link _localVideoActive}
+     */
+    modifyContents(jingleContents) {
+        const workFunction = finishedCallback => {
+            if (this._assertNotEnded('modify contents')) {
+                if (this._parseContentSenders(jingleContents)) {
+                    // Will do the sRD/sLD cycle to update SDPs and adjust
+                    // the media direction
+                    this._renegotiate()
+                        .then(finishedCallback, finishedCallback /* (error) */);
+                } else {
+                    finishedCallback();
+                }
+            } else {
+                finishedCallback();
+            }
+        };
+
+        logger.debug('Queued modify contents task');
+
+        this.modificationQueue.push(
+            workFunction,
+            error => {
+                if (error) {
+                    logger.error('modifyContents failed', error);
+                }
+            });
+    }
+
+    /**
+     *
+     * @param jingleContents
+     * @param finishedCallback
+     * @return {boolean}
+     * @private
+     */
+    _parseContentSenders(jingleContents) {
```

Comment:
I will probably have to merge this 2 methods into 1. Otherwise I won't find any suitable name